File: src/main/java/ro/nicolaemariusghergu/queryit/controller/PromotionController.java
package ro.nicolaemariusghergu.queryit.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import ro.nicolaemariusghergu.queryit.dto.PromotionDto;
import ro.nicolaemariusghergu.queryit.service.PromotionService;

import java.util.List;

import static ro.nicolaemariusghergu.queryit.BackEndApplication.FRONTEND_CORE_ADDRESS;

@CrossOrigin(FRONTEND_CORE_ADDRESS)
@RequestMapping("/api/promotions")
@Controller
public record PromotionController(PromotionService promotionService) {

    @GetMapping("/v1")
    @ResponseBody
    public ResponseEntity<List<PromotionDto>> getPromotions() {
        return promotionService.getPromotions();
    }

    @GetMapping("/v1/{promotionId}")
    @ResponseBody
    public ResponseEntity<PromotionDto> getPromotionById(@PathVariable Long promotionId) {
        return promotionService.getPromotionById(promotionId);
    }

    @PostMapping("/v1")
    @ResponseBody
    public ResponseEntity<Long> addPromotion(@RequestBody PromotionDto promotionDto) {
        return promotionService.addPromotion(promotionDto);
    }

    @PatchMapping("/v1")
    @ResponseBody
    public ResponseEntity<PromotionDto> updatePromotion(@RequestBody PromotionDto promotionDto) {
        return promotionService.updatePromotion(promotionDto);
    }

    @DeleteMapping("/v1/{promotionId}")
    @ResponseBody
    public ResponseEntity<Long> deletePromotion(@PathVariable Long promotionId) {
        return promotionService.deletePromotionById(promotionId);
    }
}

File: src/main/java/ro/nicolaemariusghergu/queryit/controller/PromotionController.kt
package ro.nicolaemariusghergu.queryit.controller

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.*
import ro.nicolaemariusghergu.queryit.BackEndApplication
import ro.nicolaemariusghergu.queryit.dto.PromotionDto

@CrossOrigin(BackEndApplication.FRONTEND_CORE_ADDRESS)
@RequestMapping("/api/promotions")
@Controller
class PromotionController {
    @GetMapping("/v1")
    @ResponseBody
    fun getPromotions(): ResponseEntity<MutableList<PromotionDto?>?>? {
        return promotionService.getPromotions()
    }

    @GetMapping("/v1/{promotionId}")
    @ResponseBody
    fun getPromotionById(@PathVariable promotionId: Long?): ResponseEntity<PromotionDto?>? {
        return promotionService.getPromotionById(promotionId)
    }

    @GetMapping("/v1/promotion-name")
    @ResponseBody
    fun getPromotionByName(@RequestBody promotionDto: PromotionDto?): ResponseEntity<MutableList<PromotionDto?>?>? {
        return promotionService.getPromotionsByName(promotionDto.getName())
    }

    @PostMapping("/v1")
    @ResponseBody
    fun addPromotion(@RequestBody promotionDto: PromotionDto?): ResponseEntity<Long?>? {
        return promotionService.addPromotion(promotionDto)
    }

    @PatchMapping("/v1")
    @ResponseBody
    fun updatePromotion(@RequestBody promotionDto: PromotionDto?): ResponseEntity<PromotionDto?>? {
        return promotionService.updatePromotion(promotionDto)
    }

    @DeleteMapping("/v1/{promotionId}")
    @ResponseBody
    fun deletePromotion(@PathVariable promotionId: Long?): ResponseEntity<Long?>? {
        return promotionService.deletePromotionById(promotionId)
    }
}